sale_layout_category_hide_detail/static/src/js/boolean_fa_icon_widget.js

```diff
@@ -1,101 +1,113 @@
-                    (terminology && _t(terminology.hover_false)) ||
-                    _t("Click to check");
-                var tip = this.value ? hover_true : hover_false;
-                var style = this._allowEdit() ? "" : "cursor:default";
-                // Set template and add it to $el
-                var template = "<span class='fa %s' title='%s' style='%s'></span>";
-                this.$el.empty().append(_.str.sprintf(template, fa_class, tip, style));
-            },
-
-            // --------------------------------------------------------------------------
-            // Handlers
-            // --------------------------------------------------------------------------
-
-            /**
-             * Toggle value
-             *
-             * @private
-             * @param {MouseEvent} event
-             */
-            _toggleValue: function (event) {
-                event.preventDefault();
-                event.stopPropagation();
-                if (this._allowEdit()) this._setValue(!this.value);
-            },
+/* Copyright 2022 Tecnativa - Víctor Martínez
+/* Copyright 2023 Tecnativa - Yadier Quesada
+/* License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).*/
+
+const {useState} = owl;
+import {BooleanField} from "@web/views/fields/boolean/boolean_field";
+import {_lt} from "@web/core/l10n/translation";
+import {registry} from "@web/core/registry";
+
+const iconTrue = "fa-check-square-o";
+const iconFalse = "fa-square-o";
+const tooltipTrue = "Switch to hidden";
+const tooltipFalse = "Switch to show";
+
+export class BooleanFaIconWidget extends BooleanField {
+    setup() {
+        super.setup();
+        this.state = useState({
+            fa_class: this.getFaIconClass(this.props.value),
+            text_tooltip: this.getTextTooltip(this.props.value),
         });
+    }
+
+    getFaIconClass(currentValue) {
+        var fa_icons = this.props.fa_icons;
+        var icon_true = fa_icons.icon_true || iconTrue;
+        var icon_false = fa_icons.icon_false || iconFalse;
+        return currentValue ? icon_true : icon_false;
+    }
+
+    getTextTooltip(currentValue) {
+        var show_tooltip = this.props.terminology;
+        var tooltip_true = show_tooltip.hover_true || tooltipTrue;
+        var tooltip_false = show_tooltip.hover_false || tooltipFalse;
+        return currentValue ? _lt(tooltip_true) : _lt(tooltip_false);
+    }
+
+    isAllowEdit() {
+        var resModel = this.props.record.resModel;
+        var state = this.props.record.data.state;
+        if (typeof state === "undefined" || state === null) {
+            state = this.props.record.data.parent_state;
+        }
+        if (
+            ["account.move.line", "sale.order.line"].includes(resModel) &&
+            state !== "draft"
+        ) {
+            this.props.allow = false;
+        }
+        return this.props.allow;
+    }
+
+    isReadonly() {
+        return this.props.record.isReadonly(this.props.name);
+    }
```

Comment:
This is a getter the inherited class: https://github.com/odoo/odoo/blob/ce3e66609e70cef6801baf9dba71c84798d382c1/addons/web/static/src/views/fields/boolean/boolean_field.js#L11-L13

Anyway, it would be nice to have documentation on why this is redefined

Reply:
Documented in the code